Libs/GObject/Classes/Object.TypeDictionary.cs

```diff
@@ -53,6 +57,48 @@ internal static void Add(System.Type type, Type gtype)
                 GTypeDict.Add(gtype, type);
             }
 
+            /// <summary>
+            /// Recursively register <c>type</c> in the type dictionary. Prefer
+            /// <see cref="AddRecursive(System.Type, Type)"/> if the gtype is already known.
+            /// </summary>
+            /// <param name="type">The C# type.</param>
+            internal static void AddRecursive(System.Type type)
+            {
+                Type gtype = GetTypeDescriptor(type)!.GType;
```

Comment:
Changed this to throw a more descriptive exception (as a type must always possess a TypeDescriptor to be registered), rather than just a null pointer exception. Prompts the user to check if they've passed a valid GObject, or some other type by accident.